applauncher/001-three-icons-horizontal.qml

```diff
@@ -0,0 +1,163 @@
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the author nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import QtQuick 2.9
+import org.asteroid.controls 1.0
+
+ListView {
+    id: appsListView
+    orientation: ListView.Horizontal
+    snapMode: ListView.SnapToItem
+    anchors.fill: parent
+    clip: true
+
+    preferredHighlightBegin: width /2 - currentItem.width /2
+    preferredHighlightEnd: width /2 + currentItem.width /2
+    highlightRangeMode: ListView.StrictlyEnforceRange
+    contentY: -(width / 2 - (width / 4))
+
+    property int currentPos: 0
+
+    onCurrentPosChanged: {
+        rightIndicator.animate()
+        leftIndicator.animate()
+        topIndicator.animate()
+        bottomIndicator.animate()
+    }
+
+    Connections {
+        target: grid
+        function onCurrentVerticalPosChanged() {
+            if (grid.currentVerticalPos == 1) {
+                grid.changeAllowedDirections()
+            }
+        }
+    }
+
+    model: launcherModel
+
+    delegate: MouseArea {
+        id: launcherItem
+        width: appsListView.width / 2
+        height: appsListView.width / 2
+        enabled: !appsListView.dragging
+
+        onClicked: model.object.launchApplication()
+
+        Image {
+            anchors.fill: parent
+            source: launcherItem.pressed | fakePressed ? "../applauncher-img/diskBackgroundPressed.svg" : "../applauncher-img/diskBackground.svg"
+            sourceSize.width: width
+            sourceSize.height: height
+        }
+
+        Icon {
+            id: icon
+            anchors.centerIn: parent
+            anchors.verticalCenterOffset: -Dims.h(1)
```

Comment:
I wonder why we keep this offset ? It makes sense in the default style where the text is included below the icon in the disk but here since the icon is the only thing in the disk, it would maybe look better to center the icon ?

Reply:
I don't think there is a particular reason for this vertical offset. Probably some artifact from the default app launcher style.